test_api.py

```diff
@@ -1912,33 +1921,32 @@ def _delete(self, url_query="", header=None):
             with patch("app.events.datetime", **{"utcnow.return_value": self.timestamp}):
                 url = f"{self.delete_url}{url_query}"
                 self.delete(url, 200, header, return_response_as_json=False)
-                event, key = m.events[0]
-                return json.loads(event), key
+                return m.events[0]
 
     def test_create_then_delete(self):
         self._check_hosts_are_present((self.host_to_delete.id,))
-        event, key = self._delete()
-        self._assert_event_is_valid(event, key, self.host_to_delete, self.timestamp)
+        event = self._delete()
+        self._assert_event_is_valid(event, self.host_to_delete, self.timestamp)
         self._check_hosts_are_deleted((self.host_to_delete.id,))
 
     def test_create_then_delete_with_branch_id(self):
         self._check_hosts_are_present((self.host_to_delete.id,))
-        event, key = self._delete(url_query="?branch_id=1234")
-        self._assert_event_is_valid(event, key, self.host_to_delete, self.timestamp)
+        event = self._delete(url_query="?branch_id=1234")
+        self._assert_event_is_valid(event, self.host_to_delete, self.timestamp)
         self._check_hosts_are_deleted((self.host_to_delete.id,))
 
     def test_create_then_delete_with_request_id(self):
         request_id = generate_uuid()
         header = {"x-rh-insights-request-id": request_id}
-        event, key = self._delete(header=header)
-        self._assert_event_is_valid(event, key, self.host_to_delete, self.timestamp)
-        self.assertEqual(request_id, event["request_id"])
+        event = self._delete(header=header)
+        self._assert_event_is_valid(event, self.host_to_delete, self.timestamp)
+        self.assertEqual(request_id, event.value["request_id"])
 
     def test_create_then_delete_without_request_id(self):
         self._check_hosts_are_present((self.host_to_delete.id,))
-        event, key = self._delete(header=None)
-        self._assert_event_is_valid(event, key, self.host_to_delete, self.timestamp)
-        self.assertEqual("-1", event["request_id"])
+        event = self._delete(header=None)
```

Comment:
```suggestion
        self._delete()

        emit_event.assert_called_once()
        event = emitted_event(emit_event.mock_calls[0])
```

I know this is not very DRY, but these are tests. They need to be straightforward. The closer are the mocks, patches and asserts to the actual test classes/methods, the better.